File: spec/cypress/integration/posts_spec.js
describe("Post Creation", function() {
    beforeEach(function() {
        cy.appScenario('basic_user').then(user => {
            cy.login(user);
        });
    })

    it('works with rich text', function() {
        cy.get("h1 a[title='Add Post']").click()

        cy.get('#post_title').type('The Happy Beaver Society');
        cy.get('#post_body').type('Happy Beavers everywhere - ');
        cy.get('button[title=Bold]').click();
        cy.get('#post_body').type('Look out!');
        cy.get('#post_body').attachFile('images/beaver_happy.png', { subjectType: 'drag-n-drop' });

        // Necessary for the upload to happen...
        cy.wait(2000);

        cy.get("input[type=submit]").click();

        cy.contains('.card-header', 'The Happy Beaver Society').should('exist')
    })
})

File: spec/features/posts_spec.rb
RSpec.describe "Post Creation" do
  let(:current_user) { create(:user) }

  before(:each) do
    login(current_user)
    visit(posts_path)
  end

  it "works" do
    find("h1 a[title='Add Post']").click
    find("#post_title").fill_in(with: "The Happy Beaver Society")
    # find("#post_body").fill_in(with: "Happy Beavers everywhere - ")
    find("button[title=Bold]").click

    # find("#post_body").fill_in(with: "Look Out")

    page.drop_file("#post_body", Rails.root.join("spec", "cypress", "fixtures", "images", "beaver_happy.png"))
    sleep(2)

    find("input[type=submit]").click

    expect(find(".card-header")).to have_text("The Happy Beaver Society")
  end
end
